src/integrations/ga/SplitToGa.js

```diff
@@ -1,10 +1,108 @@
+   * and logged (as warnings or errors depending the case) by GA debugger, but the hit is sent anyway.
+   *
+   * @param {UniversalAnalytics.FieldsObject} fieldsObject object to validate.
+   * @returns {boolean} Whether the data instance is a valid FieldsObject or not.
+   */
+  static validateFieldsObject(fieldsObject) {
+    if (fieldsObject && fieldsObject.hitType)
+      return true;
+
+    log.warn('your custom mapper returned an invalid FieldsObject instance. It must be an object with at least a `hitType` field.');
+    return false;
+  }
+
+  constructor(options) {
+
+    // Check if `ga` object is available
+    if (typeof SplitToGa.getGa() !== 'function') {
+      // @TODO review the following warning message
+      log.warn('`ga` command queue not found. No hits will be sent.');
+      // Return an empty object to avoid creating a SplitToGa instance
+      return {};
+    }
+
+    this.filter = options && typeof (options.filter) === 'function' ?
+      options.filter :
+      SplitToGa.defaultFilter;
+
+    // @TODO Should we check something else about `configObject.impressionMapper`?
+    // It doesn't matter, because if the returned object is not a GA fieldsObject or string, ga send command will do nothing.
+    this.mapper = options && typeof (options.mapper) === 'function' ?
+      options.mapper :
+      SplitToGa.defaultMapper;
+
+    this.trackerNames = options && Array.isArray(options.trackerNames) ?
+      // We strip off duplicated values if we received a `trackerNames` param.
+      // We don't warn if a tracker does not exist, since the user might create it after the SDK is initialized.
+      // Note: GA allows to create and get trackers using a string or number as tracker name, and does nothing if other types are used.
+      uniq(options.trackerNames) :
+      SplitToGa.defaultTrackerNames;
+  }
+
+  queue(data) {
+    try {
+      // filter and map Split events/impressions into a FieldsObject instance
+      const fieldsObject = this.filter(data) && this.mapper(data);
+
+      // don't send the hit if it is falsy or invalid when generated by a custom mapper
+      if (!fieldsObject || (this.mapper !== SplitToGa.defaultMapper && !SplitToGa.validateFieldsObject(fieldsObject)))
+        return;
+
+      // send the hit
+      this.trackerNames.forEach(trackerName => {
+        const sendCommand = trackerName ? `${trackerName}.send` : 'send';
+        // access ga command queue via `getGa` method, accounting for the possibility that
+        // the global `ga` reference was not yet mutated by analytics.js.
+        SplitToGa.getGa()(sendCommand, fieldsObject);
+      });
+    } catch (err) {
+      log.warn(`SplitToGa queue method threw: ${err}. No hit was sent.`);
+    }
+  }
+
 }
 
-export default SplitToGaFactory;
\ No newline at end of file
+// A falsy object represents the default tracker
+SplitToGa.defaultTrackerNames = [''];
```

Comment:
Why not to have this on the class definition?

Reply:
`static defaultTrackerNames = [''];` is not a valid JS syntax :/